force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -66,20 +126,43 @@ public with sharing class DummyJSONCallout {
 	@TestVisible // Allows test class to see this method. Since it is private, it would not be visible otherwise.
 	private static Contact createContactFromJson(String jsonResponse) {
 		// Deserialize the JSON string into map of primitive data types.
-
+		Map<String, Object> parsedResponse = (Map<String, Object>) JSON.deserializeUntyped(jsonResponse);
 		// Create a new contact from the JSON response.
-
-		// Set the contact fields from the JSON response.
+		Contact cont = new Contact();
+		// Set the contact fields from the JSON response. email, phone, birthdate
+			cont.FirstName = (String) parsedResponse.get('firstName');
+			cont.LastName = (String) parsedResponse.get('lastName');
+			cont.Email = (String) parsedResponse.get('email');
+			cont.Phone = (String) parsedResponse.get('phone');
+			cont.Birthdate = Date.valueOf((String) parsedResponse.get('birthDate'));
 
 		// Deserialize the address from the JSON response.
-
-		// Set the address fields on the contact.
-
-		return null;
+		Map<String, Object> parsedFullAddress = (Map<String, Object>) parsedResponse.get('address');
+		// Set the address fields on the contact: MailingStreet, MailingCity, MailingPostalCode, MailingState, MailingCountry
+			cont.MailingStreet = (String) parsedFullAddress.get('address');
+			cont.MailingCity = (String) parsedFullAddress.get('city');
+			cont.MailingPostalCode = (String) parsedFullAddress.get('postalCode');
+			cont.MailingState = (String) parsedFullAddress.get('state');
+			cont.MailingCountry = (String) parsedFullAddress.get('country');
+		
+			//String DummyJsonLastUpdatedString = Datetime.now().format('yyyy-MM-dd');
+			//String DummyJsonLastUpdatedString = Date.today().format();
+		// Get the current Datetime
+	Datetime myDateTime = Datetime.now();
+
+	// Convert Datetime to String in the desired format (e.g., ISO 8601)
+	//String DummyJsonLastUpdatedString = myDateTime.format('yyyy-MM-dd\'T\'HH:mm:ss\'Z\'');
+
+	// Assign the formatted String to the text field
+	cont.DummyJSON_Last_Updated__c = myDateTime;
+
+	Upsert cont;
```

Comment:
This upsert here is not needed, you can just return the contact and use the upsert on line 101